crates/chain/src/local_chain.rs

```diff
@@ -205,18 +224,39 @@ pub struct Update {
 #[derive(Debug, Clone)]
 pub struct LocalChain {
     tip: CheckPoint,
-    index: BTreeMap<u32, BlockHash>,
+    xor: BlockHashXor,
```

Comment:
I think this is a bit premature. I don't know that we need really fast equality for anything.  We didn't have fast equality before. If we were to do it, I'd just do it in `CheckPoint`.

Reply:
> If we were to do it, I'd just do it in `CheckPoint`.

Do you mean implement fast equality in `CheckPoint`? So every checkpoint will have an `xor` field?

Reply:
Yes, I agree that adding the xor right now is premature.

Reply:
> > If we were to do it, I'd just do it in `CheckPoint`.
> 
> Do you mean implement fast equality in `CheckPoint`? So every checkpoint will have an `xor` field?

Yes. As discussed in #1380 we probably do want to do this for `CheckPoint` in the future.